src/items.rs

```diff
@@ -1220,25 +1220,40 @@ impl Rewrite for ast::StructField {
         }
 
         let type_annotation_spacing = type_annotation_spacing(context.config);
-        let result = match name {
-            Some(name) => {
-                format!("{}{}{}{}:{}",
-                        attr_str,
-                        vis,
-                        name,
-                        type_annotation_spacing.0,
-                        type_annotation_spacing.1)
-            }
+        let mut result = match name {
+            Some(name) => format!("{}{}{}{}:", attr_str, vis, name, type_annotation_spacing.0),
             None => format!("{}{}", attr_str, vis),
         };
 
-        let last_line_width = last_line_width(&result);
+        let last_line_width = last_line_width(&result) + type_annotation_spacing.1.len();
         let budget = try_opt!(shape.width.checked_sub(last_line_width));
-        let rewrite = try_opt!(self.ty
-                                   .rewrite(context,
-                                            Shape::legacy(budget,
-                                                          shape.indent + last_line_width)));
-        Some(result + &rewrite)
+        let type_rewrite = self.ty
+            .rewrite(context,
+                     Shape::legacy(budget, shape.indent + last_line_width));
+        match type_rewrite {
+            Some(ty) => {
+                if name.is_some() {
+                    result.push_str(type_annotation_spacing.1);
+                }
+                Some(result + &ty)
+            }
+            None => {
+                let type_offset = shape.indent.block_indent(context.config);
+                let budget = try_opt!(context
+                                          .config
+                                          .max_width
+                                          .checked_sub(type_offset.width()));
+                let type_rewrite = try_opt!(self.ty
+                                                .rewrite(context,
+                                                         Shape::legacy(budget, type_offset))
+                                                .map(|s| {
+                                                         format!("\n{}{}",
+                                                  type_offset.to_string(&context.config),
+                                                  s)
+                                                     }));
+                Some(result + &type_rewrite)
```

Comment:
I think this code could be simpler, rather than map the rewrite, you could make the returned value `Some(format!({}\n{}{}), result, type_offset.to_string(...), type_rewrite)`